app/assets/javascripts/custom_quick_search.js.erb

```diff
@@ -1,6 +1,88 @@
-//= require quick_search/google_analytics
-//= require quick_search/event_tracking
-//= require quick_search/xhr_search
-//= require quick_search/quicksearch
-//= require quick_search/appstats
-<%= require_asset(QuickSearch::Engine::APP_CONFIG['theme']) %>
+function remove_timedout_spinners(){
+  $('.fa-refresh').not('.icon-5x').remove();
+  $('#no-results-trying-again').remove();
+}
+
+function add_found_item_types(endpoint) {
+  $('.result-types li.' + endpoint + " .no-results-label")
+    .replaceWith('<a href="#' + endpoint + '" data-quicksearch-ga-action="' +
+      endpoint + '">' + I18n["en"][endpoint + "_search"]["display_name"] + ' </a>');
+}
+
+var xhr_searches = function(){
+  // We want to remove the spinners once all of the ajax requests are done. To do this we create an array of all of the ajax
+  // requests, which are promises. We can then apply this array so that all the promises are wrapped up together. When
+  // they're all done then the 'then' function runs removing the spinners.
+  $.when.apply(this,
+    $('.search-error').map(function(index){
+      var that = $(this);
+      var parent = that.parent('.module-contents');
+      var url = that.data('quicksearch-xhr-url');
+
+      parent.append('<div class="trying_again"><i class="fa fa-refresh fa-spin"></i> Just a few more seconds </div>');
```

Comment:
I recently found out about Bootstrap 4's spinners. https://getbootstrap.com/docs/4.5/components/spinners/ just mentioning it here if we ever want to get out of FontAwesome land.